### ISSUE TITLE
RDS checks do not work with terraform-aws-modules/rds/aws module (nested modules issue?)

### ISSUE THREAD

**Issue by lboynton** [tfsec/tfsec#549]
**Describe the bug**
The RDS checks in tfsec do not detect issues in usage of the RDS module available at: https://registry.terraform.io/modules/terraform-aws-modules/rds/aws/latest

**To Reproduce**
Steps to reproduce the behaviour:
1. Create the following terraform file:
```terraform
module "db" {
  source = "terraform-aws-modules/rds/aws"
  version = "~> 2.0"

  identifier = "demodb"

  engine            = "mysql"
  engine_version    = "5.7.19"
  instance_class    = "db.t2.large"
  allocated_storage = 5

  name     = "demodb"
  username = "user"
  password = aws_ssm_parameter.pw
  port     = "3306"

  maintenance_window = "Mon:00:00-Mon:03:00"
  backup_window      = "03:00-06:00"
}

resource "aws_ssm_parameter" "pw" {
  name = "pw"
  type = "SecureString"
  value = "changeme"
}
```
2. Run `tfsec .` in the directory containing the file
3. No issues detected

**Expected behavior**
I would expect this to detect that the RDS instance is not encrypted, because encryption is not enabled by default.

**Additional context**
I think this is caused by the fact that the RDS module uses some modules that it defines itself. If you call the modules it provides directly, tfsec does detect issues:

```terraform
module "db_instance" {
  source = "./.terraform/modules/db/modules/db_instance"

  identifier = "demodb"

  engine            = "mysql"
  engine_version    = "5.7.19"
  instance_class    = "db.t2.large"
  allocated_storage = 5

  name     = "demodb"
  username = "user"
  password = aws_ssm_parameter.pw
  port     = "3306"

  maintenance_window = "Mon:00:00-Mon:03:00"
  backup_window      = "03:00-06:00"
}

resource "aws_ssm_parameter" "pw" {
  name = "pw"
  type = "SecureString"
  value = "changeme"
}
```
Results in what you'd expect:
```text
Problem 1

  [AWS052][ERROR] Resource 'module.db_instance:aws_db_instance.this' defines a disabled RDS Instance Encryption.
  /home/me/tfsec-rds-check/.terraform/modules/db/modules/db_instance/main.tf:50

      47 |   allocated_storage = var.allocated_storage
      48 |   storage_type      = var.storage_type
      49 |   storage_encrypted = var.storage_encrypted
      50 |   kms_key_id        = var.kms_key_id    [string] ""
      51 |   license_model     = var.license_model
      52 | 
      53 |   name                                = var.name

   See https://tfsec.dev/docs/aws/AWS052/ for more information.
```

**Comment by codepay4999** [comment]
good issue! I spent around 2 hours figuring out what's wrong with my local setup! I think I will look into this when I get a chance

**Comment by owenrumney** [comment]
Hey @lboynton  - this is an interesting one. I've got a PR that will do the deeper scan, processing the modules and submodules found in following a `terraform init`

I'm worried this will generate too much info, I've attached the output of your example, would be great to get your thoughts/reaction on seeing it https://gist.github.com/owenrumney/cfb60e085217623bd3b41cce74ce3f26

**Comment by owenrumney** [comment]
I'm wondering about adding a `--exclude-downloaded-modules` to remove them from the results?

**Comment by owenrumney** [comment]
@lboynton - this has been released as v0.37.0

**Comment by lboynton** [comment]
Thanks! Will give it a try over the next few days.

I think your approach makes sense, I'd rather have duplicated output from tfsc instead of a false sense of security with no output. :+1:
